src/components/generator/GenerateDialog.js

```diff
@@ -0,0 +1,98 @@
+import Notifications from "../common/Notifications";
+import {Button, Dialog, DialogContent, DialogTitle, IconButton, styled} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import Settings from "../common/settings";
+import { useEffect, useState } from "react";
+import FetchHandler from "../common/FetchHandler";
+
+const GenerateDialogRoot = styled('div')({
+  width: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+
+  '> *': {
+    marginBottom: '16px',
+  }
+});
+
+const StyledButton = styled(Button)({
+    cursor: 'pointer',
+    width: '128px',
+    display: 'flex',
+
+    ':hover': {
+        backgroundColor: '#BFBFBF',
+    },
+});
+
+const StyledA = styled('a')({
+  textDecoration: "none",
+});
+
+const StyledIconButton = styled(IconButton)({
+    position: 'absolute',
+    top: 10,
+    right: 0,
+});
+
+const StyledDialogContent = styled(DialogContent)({
+    display: "flex",
+    justifyContent: "center"
+});
+
+const EditDialog = ({setGenerate, open, setOpen, selectedProject, setSelectedProject}) => {
+    const [projectRetrievalState, setProjectRetrievalState] = useState("loading");
+
+    useEffect(() => {
+        FetchHandler.simple(fetch(Settings.generatePath + selectedProject.id, {method: "GET"}))
+            .then(() => {
+                setProjectRetrievalState("success");
+            })
+            .catch((err) => {
+                Notifications.notify(`Failed to retrieve data\n${err}`, "error");
+                setProjectRetrievalState("failed");
+            });
+      }, [selectedProject]);
```

Comment:
I don't think we need this (nor do we need the `projectRetrievalState`).

Reply:
This enables the download button. I'd like to keep that functionality. Do you have a workaround in mind?

Reply:
Do I understand correctly that in a happy case workflow the backend receives the exact same request at the generation endpoint twice and only the second time the local download is actually triggered?

If that's the case then I'd rather that we implemented something like a `/project/generate/validate` endpoint yielding just `false` or `true`. We shouldn't fire up the template engine and do zip compression just to check whether the project is valid. This will be especially bad in a project with more than a few dozen welding points.

I opened a [backend PR](https://github.com/SQE22-RoboticWelding/model-transformer-welding-backend/pull/65) providing just the validation part.

Reply:
alright, i'll be waiting for that. :)

Reply:
The backend PR is mrged now

Reply:
is implemented